docs/proposal/kmesh-reliability.md

```diff
@@ -0,0 +1,164 @@
+
+  | Scenario     | workload model | Kmesh management | Kmesh byPass | Auth | waypoint Interconnection                                     |
+  | ------------ | -------------- | ---------------- | ------------ | ---- | ------------------------------------------------------------ |
+  | Existed link | Y              | Y                | Y            | Y    | P(If the connection is complete but the send request has not been sent) |
+  | New link     | N              | N                | N            | N    | N                                                            |
+
+  Countermeasures:
+
+  - The bpf prog and map need to be maintained during restart.
+
+- After the Kmesh is restarted:
+
+  | Scenario     | workload model | Kmesh management | Kmesh byPass | Auth | waypoint Interconnection |
+  | ------------ | -------------- | ---------------- | ------------ | ---- | ------------------------ |
+  | Existed link | Y              | Y                | Y            | Y    | Y                        |
+  | New link     | Y              | N                | Y            | N    | N                        |
+
+  Countermeasures:
+
+  - The bpf prog and map need to be maintained during restart.
+  - After the Kmesh is started, the data in the map needs to be restored and the data needs to be synchronized with the control plane.
+
+  Note: Currently, Kmesh supports only link-level authentication.
+
+#### Solutions
+
+![](pics/reliability-arch.svg)
+
+- eBPF management
+
+  - To keep the governance capability during the Kmesh restart, you need to place the bpf prog and bpf map pin to the /sys/fs/bpf directory on the node.
+  - During the startup of the Kmesh, the BPF PROG is replaced and the BPF MAP is inherited.
+
+- Configure recovery
+
+  - Workload resources
+    - During the startup of the Kmesh, the local bpf map data is restored to the user-mode cache before the Kmesh subscribes to workload model data from the Istiod.
+    - Incremental subscription to Istiod based on workload resources in the local cache;
+  - Kmesh management/byPass
+    - Move the management logic from kmesh-plg to kmesh-daemon.
+    - After the restart, synchronize the configuration based on the information restored by the bpf map and the annotation information on the subscribed pod. (The data needs to be compared by the Kmesh.)
+  - Authentication
+    - Restore the user-mode authentication cache information based on the authentication rule delivered by the Istiod.
+  - Waypoint interconnection
+    - Restore the user-mode authentication cache information based on the authentication rule delivered by the Istiod. The procedure is the same as that for restoring workload resources.
+
+- Resource recycling
+
+  In the scenario where the Kmesh is uninstalled, the bpf prog and bpf map need to be cleared.
+
+  - Postponed clearance
+
+    When the Kmesh exits, start a delay clearing task on the node. If the Kmesh restarts abnormally, delete the scheduled task after the restart.
```

Comment:
ALternatively we can detect whether there is a new kmesh daemon pod in the same node. And only clear the resources if no such pod exists